contracts/src/contracts/hooks/aggregation_hook.cairo

```diff
@@ -0,0 +1,92 @@
+#[starknet::contract]
+pub mod aggregation_hook {
+    use alexandria_bytes::{Bytes, BytesTrait};
+    use hyperlane_starknet::contracts::hooks::libs::standard_hook_metadata::standard_hook_metadata::{
+        StandardHookMetadata, VARIANT,
+    };
+    use hyperlane_starknet::contracts::libs::message::Message;
+    use hyperlane_starknet::interfaces::{
+        IPostDispatchHook, Types, IPostDispatchHookDispatcher, IPostDispatchHookDispatcherTrait
+    };
+    use starknet::{ContractAddress};
+
+    #[storage]
+    struct Storage {
+        hooks: LegacyMap::<usize, ContractAddress>,
+        hook_count: usize,
+    }
+
+    mod Errors {
+        pub const INVALID_METADATA_VARIANT: felt252 = 'Invalid metadata variant';
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, hooks: Span<ContractAddress>) {
+        let mut i = 0;
+        loop {
+            if i >= hooks.len() {
+                break;
+            }
+
+            self.hooks.write(i, *hooks.at(i));
+            i += 1;
+        };
+
+        self.hook_count.write(hooks.len());
+    }
+
+    #[abi(embed_v0)]
+    impl IPostDispatchHookImpl of IPostDispatchHook<ContractState> {
+        fn hook_type(self: @ContractState) -> Types {
+            Types::AGGREGATION(())
+        }
+
+        fn supports_metadata(self: @ContractState, _metadata: Bytes) -> bool {
+            _metadata.size() == 0 || StandardHookMetadata::variant(_metadata) == VARIANT.into()
+        }
+
+        fn post_dispatch(
+            ref self: ContractState, _metadata: Bytes, _message: Message, _fee_amount: u256
+        ) {
+            assert(self.supports_metadata(_metadata.clone()), Errors::INVALID_METADATA_VARIANT);
+
+            let hook_count = self.hook_count.read();
+            let mut i = 0_usize;
+            loop {
+                if i >= hook_count {
+                    break;
+                }
+
+                let contract_address = self.hooks.read(i);
+
+                IPostDispatchHookDispatcher { contract_address }
```

Comment:
Here it is required to call the hook quote dispatch before calling the post dispatch, in order to retrieve the amount to send to the hook contract

Reply:
Thanks for your review !
I've added quote/funds checking as discussed on TG.